### DISCUSSION TITLE
ApolloConfigChangeListener中使用setNewValue 不生效

### DISCUSSION THREAD

**Discussion by 442803117** [apolloconfig/apollo#4870] [General]
// 需求：在监听到修改的属性值是P[ 开头的时候，想把该值修改为OPT开头的新值
// 代码如下，但是实际并未修改成功，想知道如何实现
@ApolloConfigChangeListener
private void onChange(ConfigChangeEvent changeEvent) {
            for (String key : changeEvent.changedKeys()) {
            ConfigChange change = changeEvent.getChange(key);
            if (change.getNewValue().startsWith("P[") &&
                    change.getNewValue().endsWith("]")) {
                // 满足条件的场合，修改为新的值
                change.setNewValue("OPT" + change.getNewValue());
            }
        }

}

**Comment by nobodyiam** [comment]
ConfigChangeEvent 只是事件，对实际的配置没有影响的，如果要修改配置的话，可以参考 https://github.com/apolloconfig/apollo/issues/1459#issuecomment-1114440269
